analyzer/java/com/engflow/bazel/invocation/analyzer/suggestionproviders/NoCacheActionsSuggestionProvider.java

```diff
@@ -73,6 +80,23 @@ public SuggestionOutput getSuggestions(DataManager dataManager) {
       if (actionsWithoutRemoteCacheCheck.isEmpty()) {
         return noSuggestions();
       }
+      var longEnoughActions =
+          actionsWithoutRemoteCacheCheck.stream()
+              .filter(action -> minDuration.compareTo(action.getAction().duration) <= 0)
+              .collect(Collectors.toList());
+      if (longEnoughActions.isEmpty()) {
+        return SuggestionProviderUtil.createSuggestionOutput(
+            ANALYZER_CLASSNAME,
+            null,
+            ImmutableList.of(
+                SuggestionProviderUtil.createCaveat(
+                    String.format(
+                        "No actions that are not using remote caching were highlighted. None of the"
+                            + " %d actions that do not check the remote cache took sufficiently"
+                            + " long.",
```

Comment:
nit:

```suggestion
                        "%d actions do not check the remote cache, but none are listed here because all of them were fast."
```

Reply:
As no suggestion is generated, this is a top-level caveat. It needs a bit more context than caveats attached to a suggestion. How about:

> %d actions do not check the remote cache, but none were highlighted in a suggestion, because none took longer than 5 s.

I'm weary of saying "fast", it's very relative.

Reply:
> %d actions do not check the remote cache, but none were highlighted in a suggestion, because none took longer than 5 s.

^ SGTM.